### PR TITLE
Add vnet and environment setting to Pipeline Witness bicep

### PR DESCRIPTION
**Pull request by hallipr**

### CODE REVIEW

**Comment by hallipr** [comment]
/check-enforcer override

Staging deployment succeeded:
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3946671&view=logs&s=0ae12840-2c6c-5dd4-a8b4-db39bb7e7a42

File: tools/pipeline-witness/infrastructure/bicep/appResourceGroup.bicep [hunk 1]
```diff
@@ -1,11 +1,73 @@
...
+  }
+}
+
+resource vnet 'Microsoft.Network/virtualNetworks@2023-11-01' = {
+  name: vnetName
+  location: 'westus2'
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        '10.8.0.0/16'
```
Comment (mikeharder) [inline tools/pipeline-witness/infrastructure/bicep/appResourceGroup.bicep:26]: Why are you adding only this prefix?

**Review by mikeharder — CHANGES_REQUESTED** [review]
Why do we need an IP allow-list?

File: tools/pipeline-witness/infrastructure/bicep/appResourceGroup.bicep (hunk 1 above)
Comment (hallipr) [reply tools/pipeline-witness/infrastructure/bicep/appResourceGroup.bicep:26]: not sure.  Was just in the import.   Should I remove it?

**Review by mikeharder — APPROVED** [review]

**Review by weshaggard — APPROVED** [review]

**Pull request merged**